Error compiling gazebo9 ModelTreeWidget.cc

Hi, I'm installing gazebo9 (for use with px4-sim).

I ran `brew install osrf/homebrew-simulation/gazebo9 --verbose` to get this build log:

https://gist.github.com/egradman/3aabc6b4d6a3242ba81d4cdb9d87bc55

Here's the good (bad) bits:

```
/tmp/gazebo9-20200912-95689-1ea5hlg/gazebo-9.14.0/gazebo/gui/model/ModelTreeWidget.cc:165:56: error: use of undeclared identifier '_1'
      boost::bind(&ModelTreeWidget::OnSaveModel, this, _1)));
```

--- a/Formula/gazebo10.rb
+++ b/Formula/gazebo10.rb
@@ -52,6 +52,12 @@ class Gazebo10 < Formula
   conflicts_with "gazebo11", because: "differing version of the same formula"
 
   patch do
+    # Fix for compatibility with boost 1.73
+    url "https://github.com/osrf/gazebo/commit/c55942ed348580317ea77312f7efce5c7937a49c.diff?full_index=1"
+    sha256 "3da0aa653a5ffeff46cffd7c8ab84a000dfd5ad55126e7df34f8971aff3e3a9c"
+  end
+
+  patch do
     # Fix build when homebrew python is installed
     # keep this patch
     url "https://gist.githubusercontent.com/scpeters/9199370/raw/afe595587e38737c537124a3652db99de026c272/brew_python_fix.patch"

--- a/Formula/gazebo9.rb
+++ b/Formula/gazebo9.rb
@@ -51,6 +51,12 @@ class Gazebo9 < Formula
   conflicts_with "gazebo11", because: "differing version of the same formula"
 
   patch do
+    # Fix for compatibility with boost 1.73
+    url "https://github.com/osrf/gazebo/commit/c55942ed348580317ea77312f7efce5c7937a49c.diff?full_index=1"
+    sha256 "3da0aa653a5ffeff46cffd7c8ab84a000dfd5ad55126e7df34f8971aff3e3a9c"
+  end
+
+  patch do
     # Fix build when homebrew python is installed
     # keep this patch
     url "https://gist.githubusercontent.com/scpeters/9199370/raw/afe595587e38737c537124a3652db99de026c272/brew_python_fix.patch"

